src/movies.js

```diff
@@ -1,11 +1,34 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
-
+function getAllDirectors(movies){
+    let directorsarray = movies.map(function(movie){
+          return movie.director
+    })
+  return directorsarray
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+function howManyMovies(movies){
+    let SpielbergOnes = movies.filter(function(movie){
+        return movie.director == 'Steven Spielberg' && movie.genre.includes('Drama') == true;
+    })
+    return SpielbergOnes.length
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies){
+    let avrg
+    let addallscores = movies.reduce(function(acc,movie){
+       return acc + movie.rate
+    },0)
+    avrg = addallscores/movies.length.toFixed(2)
```

Comment:
the toFixed is being applied to the movies.length and not to the division. You can wrap it in paranthesis to make it work.
Also, toFixed changes the number to a string, so you will need to convert it back to a number using one of several methods (Number. , parseFloat or the + symbol)